spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -565,10 +589,96 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
+        Measure the width of the lesion along the anterior-posterior axis in the **midsagittal slice** when taking into
+        account the angle correction.
+        The width is defined as the maximum lesion width in the A-P axis across all axial slices with the lesion in
+        the midsagittal slice.
+
+        :param im_lesion_data: 3D numpy array: mask of the lesion. The orientation is assumed to be RPI (because we
+        reoriented the image to RPI using orient2rpi())
+        :param p_lst: list, pixel size of the lesion
+        :param idx: int, index of the lesion
+        """
+        # Get the midsagittal slice number (the number is RPI oriented which is fine because we reoriented the
+        # im_lesion_data to RPI using orient2rpi())
+        # Note that this number is the same across all lesions (idx) as the midsagittal slice is based on the spinal
+        # cord segmentation
+        mid_sagittal_sc_slice = self.sagittal_sc_slice
+
+        # Get all axial slices (S-I direction) with the lesion for the selected sagittal slice
+        # In other words, we will iterate through the lesion in S-I direction and compute the lesion width for each
+        # axial slice with the lesion
+        # Note: we use [1] for the S-I direction as the orientation is RPI
+        axial_lesion_slices = np.unique(np.where(im_lesion_data[mid_sagittal_sc_slice, :, :])[1])
+        # Iterate across axial slices to compute lesion width
+        lesion_width_dict = {}
+        for axial_slice in axial_lesion_slices:
+            # Get the lesion segmentation mask of the selected 2D axial slice
+            slice_lesion_data = im_lesion_data[mid_sagittal_sc_slice, :, axial_slice]
+            # Get the indices of the lesion mask for the selected axial slice to compute the lesion width.
+            # The lesion width is defined as max - min + 1
+            # Note: we intentionally use 'max - min + 1' instead of 'len(np.where(slice_lesion_data)[0])' because the
+            # 'len' approach would return the number of elements, which could be influenced, for example, by the
+            # presence of holes in the lesion mask.
+            # Context: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4617#discussion_r1744031056
+            slice_min = np.where(slice_lesion_data)[0][0]     # [0] returns the most dorsal elements
+            slice_max = np.where(slice_lesion_data)[0][-1]    # [-1] returns the most ventral elements
+            lesion_width_dict[axial_slice] = slice_max - slice_min + 1
+
+        # Get the width in mm (apply the angle correction)
+        width_cur_dict = {axial_slice: p_lst[1] * np.cos(self.angles_sagittal[axial_slice]) * lesion_width
+                          for axial_slice, lesion_width in lesion_width_dict.items()}
+
+        # Get the maximum width across all axial slices
+        max_width_axial_slice = max(width_cur_dict, key=width_cur_dict.get)
+        width_cur = width_cur_dict[max_width_axial_slice]
```

Comment:
Since you don't use the slice number `max_width_axial_slice` anywhere else, this could be simplified a bit:
```suggestion
        width_cur = max(width_cur_dict.values())
```

Reply:
good point! commiting the suggestion. thanks!